src/sp_cookie_encryption.c

```diff
@@ -201,6 +216,24 @@ PHP_FUNCTION(sp_setcookie) {
     ZVAL_LONG(&params[6], httponly);
   }
 
+  /* param[3](path) is concatenated to path= and is not filtered, we can inject
+  the samesite parameter here */
+  if (path && (!cookie_node || !cookie_node->samesite)) {
+    ZVAL_STR_COPY(&params[3], path);
+  }
+  else if (cookie_node && cookie_node->samesite) {
+    if (!path) {
+      path = zend_string_init("", 0, 0);
+    }
+    /* Concatenating everything, as is in PHP internals */
+    samesite = zend_string_extend(path, ZSTR_LEN(path) + strlen(SAMESITE_COOKIE_FORMAT) + strlen(cookie_node->samesite) + 1, 0);
```

Comment:
This is ugly. Is there an other way to create zend_string without using memcpy all the way?